azavea/tilegarden
# Feature/tiling demo

## Overview
This PR plugs mapnik into the database in order to generate map tiles, which then get served up through an API.


### Demo
[Example deployment (27 Jun 2018)](https://d28f32z0lbuk2f.cloudfront.net/tile/13/2385/3103.png)
![image](https://user-images.githubusercontent.com/16990679/41936725-21c71420-795c-11e8-802f-818185f24b87.png)
![image](https://user-images.githubusercontent.com/16990679/41936741-2e65a052-795c-11e8-87c3-1a0f0fc53109.png)


### Notes
Current development environment setup doesn't necessarily obey the quirks of the real API Gateway, tiles are served from what is basically an express server. Also, the `pwd_inlets` data set is very large on a per-tile basis which may cause the lambdas to time out. Increase the value of `TIMEOUT` in `.env` if you are consistently getting un-viewable images or 502 errors.

#### Accept headers
API Gateway currently has issues serving images that can be simply displayed in-browser, so you have to manually add an `Accept:image/png` header to your lambda requests to receive proper images. This can be done either with curl (`-H "Accept:image/png"`) or in production by setting up a proxy (e.g. with CloudFront) that adds the header to all lambda requests.


## Testing Instructions
 * Run `./scripts/update` and `./scripts/server` to run the local tileserver.
 * Open `src/demo/pwd_inlets_index.html` to generate and display tiles through Leaflet.
 * Run `./scripts/publish` (optional `--new` flag if you haven't deployed the project before). Assuming your credentials are set up properly it should deploy within about a minute.
 * Change the value of TILESERVER_PATH to load files the lambdas you just deployed.

Resolves #10, resolves #19, resolves #31, resolves #32
